### ISSUE TITLE
Compiler complains about scanning issues

### ISSUE THREAD

**Issue by kant2002** [dotnet/runtimelab#1420]
I was hoping that this will be only on Windows, but when I build on ARM and run publish I still receive following error in output
```
  Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeCoreRT
  *** Methods compiled but not scanned:
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ULMod(uint64,uint64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.LMod(int64,int64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ULDiv(uint64,uint64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.LDiv(int64,int64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowLngDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowULngDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowLngArithExc()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.IDiv(int32,int32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.UDiv(uint32,uint32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.IMod(int32,int32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.UMod(uint32,uint32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowIntDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowUIntDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowIntArithExc()
  Unhandled exception. System.Exception: Scanning failure
     at ILCompiler.Program.Run(String[] args) in ilc.dll:token 0x600010f+0x1027
     at ILCompiler.Program.Main(String[] args) in ilc.dll:token 0x6000115+0x7
```

**Comment by kant2002** [comment]
Also on Windows ILC does not produce that output under debugger, just throw Exception. Is this okay?

**Comment by jkotas** [comment]
src\coreclr\tools\aot\ILCompiler.Compiler\IL\ILImporter.Scanner.cs needs to call `_dependencies.Add(GetHelperEntrypoint(...` for mul and div operations on ARM

**Comment by MichalStrehovsky** [comment]
https://github.com/dotnet/runtimelab/pull/954 has a sample.

**Comment by kant2002** [comment]
Only thing which I do not get, is how to make that dependencies work only on ARM. Or I can add them for all platforms and it would be filtered out?
Also I assume that props and targets changes are not important, and more related to other build issues.

**Comment by MichalStrehovsky** [comment]
Yup, you can ignore the build changes; they're unrelated.

`_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.ARM` can be used to filter them so that they're only added for ARM.

They're not too harmful, in the worst case they make the resulting binaries a bit bigger. But making them conditional is good hygiene.

**Comment by jkotas** [comment]
Fixed by #1423
